### DISCUSSION TITLE
Is there some config for completion dir ?

### DISCUSSION THREAD

**Discussion by jqhr** [nushell/nushell#10907] [Q&A] [answered]
Just like `~/.config/fish/completions` in fish shell.
For my config,  I want to add completions by `use` command.
But I have to write a lot of lines like `use /path/completion1.nu`.

![图片](https://github.com/nushell/nushell/assets/23305067/0f4697e3-925c-4b2e-aef2-ef55afd0786f)

`use` command dones't support pipeline.
![图片](https://github.com/nushell/nushell/assets/23305067/9eae16d4-50fc-46cb-af85-17ff71163671)

I just want it to be simple.

**Comment by fdncred** [comment, marked as answer]
No, there is not standard completion dir. What people are supposed to do is use the `NU_LIB_DIRS` env var that is in the env.nu file. So, you put your files where nushell can find them and that is in one of the folders defined in NU_LIB_DIRS, or add a new folder to it. Then you can just do `use file`.

This is what mine looks like
```nushell
# Directories to search for scripts when calling source or use
#
# By default, <nushell-config-dir>/scripts is added
$env.NU_LIB_DIRS = [
    ($nu.config-path | path dirname | path join 'scripts')
    '/Users/fdncred/src/nu_scripts'
    ($nu.config-path | path dirname)
]
```
```
❯ $env.NU_LIB_DIRS
╭─┬──────────────────────────────────────────────────────────╮
│0│/Users/fdncred/Library/Application Support/nushell/scripts│
│1│/Users/fdncred/src/nu_scripts                             │
│2│/Users/fdncred/Library/Application Support/nushell        │
╰─┴──────────────────────────────────────────────────────────╯
```
So, any file in any of those folders can be sourced or used in the config.nu or repl without specifying the full path.

So, now in my config.nu, I do this because those folders are in my nu_scripts folder.
```nushell
# git completions and aliases
use custom-completions/git/git-completions.nu *
use aliases/git/git-aliases.nu
```
